Boolean.CSharp.Test/CoreTests.cs

```diff
@@ -1,24 +1,149 @@
-﻿using Boolean.CSharp.Main;
+﻿using System.Runtime.InteropServices;
+using Boolean.CSharp.Main;
 using NUnit.Framework;
 
 namespace Boolean.CSharp.Test
 {
     [TestFixture]
-    public class CoreTests
+    public class CoreTest
     {
-        private Core _core;
+        [TestCase("SavingsAccount", 4000)]
+        [TestCase("", 4000)]
+        [TestCase("SavingsAccount", -1000)]
+        [TestCase("SavingsAccount", 0)]
+        public void TestCreateCurrentAccount(string name, decimal startingBalance) {
 
-        public CoreTests()
-        {
-            _core = new Core();
+            Branch br = new Branch("LA");
+            Account account = null;
+            TestDelegate createAccount = () => account = new CurrentAccount(name, startingBalance, br);
 
+           
+
+            if (string.IsNullOrEmpty(name) || startingBalance < 1) {
```

Comment:
better to split this into 2 tests; one to test the throwing bit; one to test the non throwing cases

Reply:
same with all other cases below